### PR TITLE
Rules & quiz page

### PR DESCRIPTION
**Pull request by amartya-rs**
In this PR: 
added rules and quiz page

preview: https://deploy-preview-2--hh-quiz.netlify.app/

### CODE REVIEW

**Review by purvasharmadev — APPROVED** [review]
everything is looking great. UI is nice. code is clean and understandable. nice work!

File: src/context/data-context.js [hunk 1]
```diff
@@ -46,6 +46,21 @@ const DataProvider = ({ children }) => {
...
+               currentCategory: payload,
+            };
+         case "SET_USER_ANSWER":
+            return {
+               ...state,
+               userAnswer: [...state.userAnswer, payload],
+            };
          default:
             return state;
       }
```
Comment (SanketDhabarde) [inline src/context/data-context.js:66]: Suggestion: You can create a separate folder for reducers and add all your reducers there

**Review by SanketDhabarde — APPROVED** [review]
Looks good to me
- Nice UI
- Clean code
- Good use of grid-template-area

File: src/context/data-context.js (hunk 1 above)
Comment (amartya-rs) [reply src/context/data-context.js:66]: Thanks for the suggestion. Will work on it.

**Pull request merged**